### ISSUE TITLE
Some tests are too perscriptive

### ISSUE THREAD

**Issue by scottaohara** [canaxess/thisiswcag#7]
I appreciate the goal of this project, to make straight forward checks for someone to reference when reviewing their content for meeting WCAG.  But, being that the (assumed) intended audience are people who find the normative WCAG standard confusing, and want a more straight forward reference, it's important not to imply authors 'must' do something, when depending on context, that 'must' might be more of an opinion than a requirement.

For instance:
>"data tables must have a caption element with a concise description of the contents"

WCAG does not require that tables "must" have a caption.  Only if there is an element that serves as a caption _must_ it also be marked up as such, or that element need to be programmatically associated with the table to give the table its accessible name.  I'd submit there are even further nuances there.

Rather, I'd expect a test like this to be worded as "ensure that if a caption is provided for a data table, that it is programmatically associated with that table."

Another example...

>"two or more radio buttons, or checkbox controls must be grouped using a `<fieldset>`"

followed by

>"and, the `<fieldset>` must have a succinct `<legend>` element"

I understand the intent here, but these are also not actually strict requirements depending on the context of these controls, and if other markup mechanisms are available (ARIA roles and labeling). 

For instance:

```html
<label><input type=radio name=f> Sign me up for your email trash</label>
<label><input type=radio name=f> I do not wish to receive trash email</label>
```

The above passes WCAG as it would be sufficiently clear to users that they are being provided options to receive an organization's <s>email trash</s> newsletter or not.

It would _arguably be better_ if it were marked up as such:

```html
<fieldset>
  <legend>Sign up for email trash?</legend> 
  <label><input type=radio name=f> Yes</label>
  <label><input type=radio name=f> No</label>
</fieldset>
```
But the following could also a valid way to expose the information without being 'required' to use a `fieldset` or `legend`

```html
<div role=radiogroup>
  <h3>Want trash in you inbox?</h3>
  <label><input type=radio name=f>Of course! (but I'll still likely delete it)</label>
  <label><input type=radio name=f>Gross, no thank you</label>
</div>
```

## Proposed resolution

I see two suitable outcomes here to make sure that this resource can both be useful (because I think it could be), but also not spread misinformation about what WCAG actually requires.

1. Carefully review and revise tests to ensure that this resource is not being overly prescriptive in what it is implying WCAG requires.  I outlined just a few in this thread, there are many more that presently can and will be interpreted as 'the way' to meet a requirement, when other options could be available / context will determine if _anything_ is even necessary to revise.  
  The potential downside to this option is that because there can be a wide variety of ways to meet a requirement, accurately listing / alluding to all of those ways would be quite a chore.  It actually might require less descriptive/prescriptive language be used.  E.g., instead of saying "use a `ul` or `ol` to create a list, a more open-ended "ensure lists of content are programmatically exposed as such" would then mean someone using a `<menu>` or `role=list` wouldn't think they need to change their markup for no reason.   
  Which then leads to option 2:

2. Better indicate that the tests/suggestions here are at times just one of many ways that a success criterion could be met.  In other words, the introduction to this website and what the content represents is not descriptive enough.  While the site's title is "this is wcag", the content of this page is not fully reflective of that claim.  Better inform people that these are starting points, and they represent ways to meet the different WCAG SCs... but this isn't all there is to it, and again, many of these tests might have unwritten alternatives which would be perfectly conforming as well.

Regardless of the solution taken, I do realize that there is a call for the community to help make this site better (hence me opening this issue).  And that is a good thing.  However, I do think that there is a responsibility of the authors of this page to more immediately rectify any misleading information, e.g., better clarifying what are actually failures / requirements vs suggested ways to meet WCAG - the latter being what many tests on this page better represent than the former.

**Comment by rossmullen** [comment]
Thanks @scottaohara  those are very reasonable points. Whilst I don't necessarily agree the information is misleading, I do appreciate the nuances of success criteria where there are many ways to tackle, and this is ultimately a starting point which covers some of the established ways of addressing a SC.

It's trying to balance an approach where if each test is too vague to allow room for all combinations of techniques to pass it leads straight back to WCAG success criteria territory which the original intention was to explain in easier language.

I do think removing the word "must" and having clearer details clarifying this as a starting point is a very good improvement and will carry that forward.

**Comment by dimitrieh** [comment]
@rossmullen perhaps it makes sense for some rules to have an asterisk added to them that indicates the rule might have some situational aspects to it and for the user to refer to original documentation for more information.
